rpc-server/src/dispatchers/blockchain.rs

```diff
@@ -156,36 +156,25 @@ impl BlockchainInterface for BlockchainDispatcher {
     ) -> RPCResult<ExecutedTransaction, (), Self::Error> {
         if let BlockchainReadProxy::Full(blockchain) = self.blockchain.read() {
             // Get all the historic transactions that correspond to this hash.
-            let mut historic_tx_vec = blockchain.history_store.get_hist_tx_by_hash(&hash, None);
+            let mut hist_txs = blockchain.history_store.get_hist_tx_by_hash(&hash, None);
 
             // Unpack the transaction or raise an error.
-            let historic_tx = match historic_tx_vec.len() {
-                0 => {
-                    return Err(Error::TransactionNotFound(hash));
-                }
-                1 => historic_tx_vec.pop().unwrap(),
-                _ => {
-                    return Err(Error::MultipleTransactionsFound(hash));
-                }
+            let hist_tx = match (hist_txs.pop(), hist_txs.len()) {
```

Comment:
```suggestion
                        if hist_txs.len() > 1 {
                return Err(Error::MultipleTransactionsFound(hash));
            }
            let hist_tx = hist_txs
                .pop()
                .map_or(Err(Error::TransactionNotFound(hash)), |x| Ok(x))?;
```

This or something resembling this suggestion looks easier to understand than the match with the tuple. At least to me.

Reply:
Thanks! I couldn't figure out something without `unwrap` and easily understandable. I like your variant. :)

Reply:
Changed.